src/main/java/net/dv8tion/jda/api/utils/MarkdownSanitizer.java

```diff
@@ -187,6 +187,35 @@ public static String escape(@Nonnull String sequence, int ignored)
                 .withStrategy(SanitizationStrategy.ESCAPE)
                 .compute(sequence);
     }
+    
+    /**
+     * Escapes every markdown formatting found in the provided string.
+     * <br>Example: {@code escape("**Hello** ~~World~~!", MarkdownSanitizer.BOLD | MarkdownSanitizer.STRIKE)}
+     *
+     * @param  sequence
+     *         The string to sanitize
+     * @param  single
+     *         If true removes all markdown characters
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If provided with null
+     *
+     * @return The string with escaped markdown
+     */
+    @Nonnull
+    public static String escape(@Nonnull String sequence, boolean single) {
+    	if(single) {
+        	String cleaned = sequence.replaceAll("([_`~*>])", "\\\\$1");
```

Comment:
You ignore the cases where these are escaped. Regex is not capable of properly replacing these characters.

Reply:
Examples where this breaks:

`\*Hello World\*` -> `\\*Hello World\\*` (escape the already escaped markdown)
`<https://example.com/index.php?with_query=true>` -> `<https://example.com/index.php?with_query=true\>` (changes the query in a breaking way)